src/pages/sicp/subcomponents/chatbot/ChatBox.tsx

```diff
@@ -0,0 +1,183 @@
+import { Button } from '@blueprintjs/core';
+import { Buffer as NodeBuffer } from 'buffer';
+import * as React from 'react';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { useSession } from 'src/commons/utils/Hooks';
+import { SourceTheme } from 'src/features/sicp/SourceTheme';
+
+import { request } from '../../../../commons/utils/RequestHelper';
+import SICPNotes from './SicpNotes';
+
+Buffer.from('anything', 'base64');
+
+if (!(window as any).Buffer) {
+  (window as any).Buffer = NodeBuffer;
+}
```

Comment:
@Song-Mengfei could you help elaborate this? I am also confused as to what it does.

Reply:
After checking with mengfei, I realize that this is for openai to work;
but since now we wrap openai client in the backend, we no longer need this.
Will get this deleted.